src/monodroid/jni/monodroid-glue.cc

```diff
@@ -506,104 +524,106 @@ MonodroidRuntime::parse_gdb_options ()
 }
 
 #if defined (DEBUG) && !defined (WINDOWS)
-int
-MonodroidRuntime::parse_runtime_args (char *runtime_args, RuntimeOptions *options)
-{
-	char **args, **ptr;
-
-	if (runtime_args == nullptr)
-		return 1;
-
-	options->timeout_time = 0;
-
-	args = utils.monodroid_strsplit (runtime_args, ",", 0);
-
-	for (ptr = args; ptr && *ptr; ptr++) {
-		const char *arg = *ptr;
-
-		if (!strncmp (arg, "debug", 5)) {
+bool
+MonodroidRuntime::parse_runtime_args (dynamic_local_string<PROPERTY_VALUE_BUFFER_LEN> &runtime_args, RuntimeOptions *options)
+{
+	if (runtime_args.length () == 0) {
+		log_warn (LOG_DEFAULT, "runtime args empty");
+		return true;
+	}
+
+	constexpr char   ARG_DEBUG[]                   = "debug";
+	constexpr size_t ARG_DEBUG_LENGTH              = sizeof(ARG_DEBUG) - 1;
+	constexpr char   ARG_TIMEOUT[]                 = "timeout=";
+	constexpr size_t ARG_TIMEOUT_LENGTH            = sizeof(ARG_TIMEOUT) - 1;
+	constexpr char   ARG_SERVER[]                  = "server=";
+	constexpr size_t ARG_SERVER_LENGTH             = sizeof(ARG_SERVER) - 1;
+	constexpr char   ARG_LOGLEVEL[]                = "loglevel=";
+	constexpr size_t ARG_LOGLEVEL_LENGTH           = sizeof(ARG_LOGLEVEL) - 1;
+
+	bool ret = true;
+	string_segment token;
+	while (runtime_args.next_token (',', token)) {
+		if (token.starts_with (ARG_DEBUG, ARG_DEBUG_LENGTH)) {
 			char *host = nullptr;
 			int sdb_port = 1000, out_port = -1;
 
-			options->debug = 1;
-
-			if (arg[5] == '=') {
-				const char *sep, *endp;
-
-				arg += 6;
-				sep = strchr (arg, ':');
-				if (sep != nullptr) {
-					size_t arg_len = static_cast<size_t>(sep - arg);
-					size_t alloc_size = ADD_WITH_OVERFLOW_CHECK (size_t, arg_len, 1);
-					host = new char [alloc_size];
-					memset (host, 0x00, alloc_size);
-					strncpy (host, arg, arg_len);
-					arg = sep+1;
-
-					sdb_port = (int) strtol (arg, const_cast<char**> (&endp), 10);
-					if (endp == arg) {
-						log_error (LOG_DEFAULT, "Invalid --debug argument.");
-						continue;
-					} else if (*endp == ':') {
-						arg = endp+1;
-						out_port = (int) strtol (arg, const_cast<char**> (&endp), 10);
-						if ((endp == arg) || (*endp != '\0')) {
-							log_error (LOG_DEFAULT, "Invalid --debug argument.");
-							continue;
-						}
-					} else if (*endp != '\0') {
-						log_error (LOG_DEFAULT, "Invalid --debug argument.");
-						continue;
+			options->debug = true;
+
+			if (token.has_at ('=', 5)) {
+				static_local_string<SMALL_STRING_PARSE_BUFFER_LEN> hostport (token.length () - 6);
```

Comment:
similarly, where is `6` coming from? It's from `ARG_DEBUG_LENGTH+2`, but that isn't really obvious, meaning *more* manual counting.

Reply:
It's from `ARG_DEBUG_LENGTH + 1` and includes the `=`